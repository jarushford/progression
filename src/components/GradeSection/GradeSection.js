import React from 'react'
import gradeConverter from '../../assets/gradeConverter'
import '../../main.scss'

export default function GradeSection({ grade, ascents }) {
  const ascentsRender = ascents.map(ascent => {
    return (
      <article className="ascent">
        <div>
          <h2 className="ascent-name">{ascent.name}</h2>
          <h5 className="ascent-location">{ascent.location}</h5>
        </div>
        <p>{ascent.caption}</p>
      </article>
    )
  })

  return (
    <div className="grade-section">
      <h3>{gradeConverter[grade]}</h3>
      {ascentsRender}
    </div>
  )
}