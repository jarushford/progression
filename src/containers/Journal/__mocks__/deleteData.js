export const deleteDataThunk = jest.fn()