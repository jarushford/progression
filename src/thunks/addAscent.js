import { fetchAscentsThunk } from './fetchAscents'
import { clearAscents, setError } from '../actions'

export const addAscentThunk = (ascent) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/ascents/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(ascent),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add ascent')
      }
      await dispatch(clearAscents())
      dispatch(fetchAscentsThunk(ascent.user_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}