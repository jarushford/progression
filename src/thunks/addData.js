import { clearThunkHelper } from '../utils/thunkHelpers'
import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const addDataThunk = (data, type) => {
  return async (dispatch) => {
    try {
      const url = `http://localhost:3000/api/progressionusers/${type}s/new`
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not add ${type}`)
      }

      await dispatch(clearThunkHelper(type)())
      type === 'journal' || type === 'milestone'
        ? dispatch(fetchDataThunk(data.user_id, data.project_id, type))
        : dispatch(fetchDataThunk(data.user_id, '', type))

      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}