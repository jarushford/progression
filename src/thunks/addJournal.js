import { fetchJournalThunk } from './fetchJournal'
import { clearJournal, setError } from '../actions'

export const addJournalThunk = (entry) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/journal/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(entry),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add entry')
      }
      await dispatch(clearJournal())
      dispatch(fetchJournalThunk(entry.user_id, entry.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}