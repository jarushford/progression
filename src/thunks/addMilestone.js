import { fetchMilestonesThunk } from './fetchMilestones'
import { clearMilestones, setError } from '../actions'

export const addMilestoneThunk = (milestone) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/milestones/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(milestone),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add project')
      }
      await dispatch(clearMilestones())
      dispatch(fetchMilestonesThunk(milestone.user_id, milestone.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}