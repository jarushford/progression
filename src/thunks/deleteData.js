import { clearThunkHelper } from '../utils/thunkHelpers'
import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const deleteDataThunk = (item_id, user_id, project_id, type) => {
  return async (dispatch) => {
    try {
      const data = {item_id, user_id, project_id}
      let url
      type === 'journal' || type === 'milestone'
        ? url = `http://localhost:3000/api/progressionusers/${user_id}/projects/${project_id}/${type}s/${item_id}`
        : url = `http://localhost:3000/api/progressionusers/${user_id}/${type}s/${item_id}`

      const response = await fetch(url, {
        method: 'DELETE',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not remove ${type}`)
      }

      await dispatch(clearThunkHelper(type)())

      type === 'journal' || type === 'milestone'
        ? dispatch(fetchDataThunk(data.user_id, data.project_id, type))
        : dispatch(fetchDataThunk(data.user_id, '', type))

    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}