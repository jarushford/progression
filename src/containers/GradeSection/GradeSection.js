import React from 'react'
import gradeConverter from '../../utils/gradeConverter'
import { deleteDataThunk } from '../../thunks/deleteData'
import { connect } from 'react-redux'
import { uid } from 'react-uid'
import '../../main.scss'

function GradeSection({ grade, ascents, user, deleteAscent }) {
  const ascentsRender = ascents.map(ascent => {
    return (
      <article className="ascent" key={uid(ascent)}>
        <div>
          <div className="content-section">
            <h2 className="ascent-name">{ascent.name}</h2>
            <h5 className="ascent-location">{ascent.location}</h5>
          </div>
          <i className="fas fa-times" onClick={() => deleteAscent(ascent.id, user.id, null, 'ascent')}></i>
        </div>
        <p>{ascent.caption}</p>
      </article>
    )
  })

  return (
    <div className="grade-section">
      <h3>{gradeConverter[grade]}</h3>
      {ascentsRender}
    </div>
  )
}

const mapStateToProps = (state) => ({
  user: state.currentUser
})

const mapDispatchToProps = (dispatch) => ({
  deleteAscent: (item_id, user_id, project_id, type) => dispatch(deleteDataThunk(item_id, user_id, project_id, type))
})

export default connect(mapStateToProps, mapDispatchToProps)(GradeSection)