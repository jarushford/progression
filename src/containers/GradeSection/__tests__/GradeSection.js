import React from 'react'
import { shallow } from 'enzyme'
import { deleteDataThunk } from '../../../thunks/deleteData'
import { mapStateToProps, mapDispatchToProps, GradeSection } from '../GradeSection'

jest.mock('../../../thunks/deleteData')

describe('GradeSection', () => {
  let mockUser
  let mockAscents
  let mockDeleteAscent

  beforeEach(() => {
    mockUser = {
      name: 'Groot',
      email: 'groot@groot.groot',
      password: 'iamgroot',
      id: 3
    }
    mockAscents = [
      {
        name: 'Top Notch',
        location: 'RMNP, CO',
        grade: 'V13',
        id: 1
      },
      {
        name: 'Megalodon',
        location: 'Newlin, CO',
        grade: 'V7',
        id: 2
      },
      {
        name: 'Sabretooth',
        location: 'Thunder Ridge, CO',
        grade: 'V13',
        id: 3
      }
    ]
    mockDeleteAscent = jest.fn()
  })
  
  describe('GradeSection Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<GradeSection user={mockUser} ascents={mockAscents} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should delete an ascent on clicking delete', () => {
      const wrapper = shallow(<GradeSection user={mockUser} ascents={mockAscents} deleteAscent={mockDeleteAscent} />)

      wrapper.find('.fa-times').first().simulate('click')

      expect(mockDeleteAscent).toBeCalledWith(1, 3, null, 'ascent')
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = { currentUser: mockUser }
      const expected = { user: mockUser }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => [
    it('should return a props object with a method deleteAscent', () => {
      const mockDispatch = jest.fn()
      const expected = deleteDataThunk()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteAscent()

      expect(mockDispatch).toBeCalledWith(expected)
    })
  ])
})