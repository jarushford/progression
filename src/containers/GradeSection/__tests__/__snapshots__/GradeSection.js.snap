// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradeSection GradeSection Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GradeSection
    ascents={
      Array [
        Object {
          "grade": "V13",
          "id": 1,
          "location": "RMNP, CO",
          "name": "Top Notch",
        },
        Object {
          "grade": "V7",
          "id": 2,
          "location": "Newlin, CO",
          "name": "Megalodon",
        },
        Object {
          "grade": "V13",
          "id": 3,
          "location": "Thunder Ridge, CO",
          "name": "Sabretooth",
        },
      ]
    }
    user={
      Object {
        "email": "groot@groot.groot",
        "id": 3,
        "name": "Groot",
        "password": "iamgroot",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        Array [
          <article
            className="ascent"
          >
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Top Notch
                </h2>
                <h5
                  className="ascent-location"
                >
                  RMNP, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>
            <p />
          </article>,
          <article
            className="ascent"
          >
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Megalodon
                </h2>
                <h5
                  className="ascent-location"
                >
                  Newlin, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>
            <p />
          </article>,
          <article
            className="ascent"
          >
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Sabretooth
                </h2>
                <h5
                  className="ascent-location"
                >
                  Thunder Ridge, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>
            <p />
          </article>,
        ],
      ],
      "className": "grade-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Top Notch
                </h2>
                <h5
                  className="ascent-location"
                >
                  RMNP, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>,
            <p />,
          ],
          "className": "ascent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Top Notch
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    RMNP, CO
                  </h5>
                </div>,
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2
                      className="ascent-name"
                    >
                      Top Notch
                    </h2>,
                    <h5
                      className="ascent-location"
                    >
                      RMNP, CO
                    </h5>,
                  ],
                  "className": "content-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Top Notch",
                      "className": "ascent-name",
                    },
                    "ref": null,
                    "rendered": "Top Notch",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "RMNP, CO",
                      "className": "ascent-location",
                    },
                    "ref": null,
                    "rendered": "RMNP, CO",
                    "type": "h5",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-times",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "uid2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Megalodon
                </h2>
                <h5
                  className="ascent-location"
                >
                  Newlin, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>,
            <p />,
          ],
          "className": "ascent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Megalodon
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Newlin, CO
                  </h5>
                </div>,
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2
                      className="ascent-name"
                    >
                      Megalodon
                    </h2>,
                    <h5
                      className="ascent-location"
                    >
                      Newlin, CO
                    </h5>,
                  ],
                  "className": "content-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Megalodon",
                      "className": "ascent-name",
                    },
                    "ref": null,
                    "rendered": "Megalodon",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Newlin, CO",
                      "className": "ascent-location",
                    },
                    "ref": null,
                    "rendered": "Newlin, CO",
                    "type": "h5",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-times",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "uid3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div
                className="content-section"
              >
                <h2
                  className="ascent-name"
                >
                  Sabretooth
                </h2>
                <h5
                  className="ascent-location"
                >
                  Thunder Ridge, CO
                </h5>
              </div>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </div>,
            <p />,
          ],
          "className": "ascent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Sabretooth
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Thunder Ridge, CO
                  </h5>
                </div>,
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2
                      className="ascent-name"
                    >
                      Sabretooth
                    </h2>,
                    <h5
                      className="ascent-location"
                    >
                      Thunder Ridge, CO
                    </h5>,
                  ],
                  "className": "content-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sabretooth",
                      "className": "ascent-name",
                    },
                    "ref": null,
                    "rendered": "Sabretooth",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Thunder Ridge, CO",
                      "className": "ascent-location",
                    },
                    "ref": null,
                    "rendered": "Thunder Ridge, CO",
                    "type": "h5",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-times",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          Array [
            <article
              className="ascent"
            >
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Top Notch
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    RMNP, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>
              <p />
            </article>,
            <article
              className="ascent"
            >
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Megalodon
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Newlin, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>
              <p />
            </article>,
            <article
              className="ascent"
            >
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Sabretooth
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Thunder Ridge, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>
              <p />
            </article>,
          ],
        ],
        "className": "grade-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Top Notch
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    RMNP, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>,
              <p />,
            ],
            "className": "ascent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="content-section"
                  >
                    <h2
                      className="ascent-name"
                    >
                      Top Notch
                    </h2>
                    <h5
                      className="ascent-location"
                    >
                      RMNP, CO
                    </h5>
                  </div>,
                  <i
                    className="fas fa-times"
                    onClick={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2
                        className="ascent-name"
                      >
                        Top Notch
                      </h2>,
                      <h5
                        className="ascent-location"
                      >
                        RMNP, CO
                      </h5>,
                    ],
                    "className": "content-section",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Top Notch",
                        "className": "ascent-name",
                      },
                      "ref": null,
                      "rendered": "Top Notch",
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "RMNP, CO",
                        "className": "ascent-location",
                      },
                      "ref": null,
                      "rendered": "RMNP, CO",
                      "type": "h5",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-times",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "uid2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Megalodon
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Newlin, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>,
              <p />,
            ],
            "className": "ascent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="content-section"
                  >
                    <h2
                      className="ascent-name"
                    >
                      Megalodon
                    </h2>
                    <h5
                      className="ascent-location"
                    >
                      Newlin, CO
                    </h5>
                  </div>,
                  <i
                    className="fas fa-times"
                    onClick={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2
                        className="ascent-name"
                      >
                        Megalodon
                      </h2>,
                      <h5
                        className="ascent-location"
                      >
                        Newlin, CO
                      </h5>,
                    ],
                    "className": "content-section",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Megalodon",
                        "className": "ascent-name",
                      },
                      "ref": null,
                      "rendered": "Megalodon",
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Newlin, CO",
                        "className": "ascent-location",
                      },
                      "ref": null,
                      "rendered": "Newlin, CO",
                      "type": "h5",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-times",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "uid3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <div
                  className="content-section"
                >
                  <h2
                    className="ascent-name"
                  >
                    Sabretooth
                  </h2>
                  <h5
                    className="ascent-location"
                  >
                    Thunder Ridge, CO
                  </h5>
                </div>
                <i
                  className="fas fa-times"
                  onClick={[Function]}
                />
              </div>,
              <p />,
            ],
            "className": "ascent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="content-section"
                  >
                    <h2
                      className="ascent-name"
                    >
                      Sabretooth
                    </h2>
                    <h5
                      className="ascent-location"
                    >
                      Thunder Ridge, CO
                    </h5>
                  </div>,
                  <i
                    className="fas fa-times"
                    onClick={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2
                        className="ascent-name"
                      >
                        Sabretooth
                      </h2>,
                      <h5
                        className="ascent-location"
                      >
                        Thunder Ridge, CO
                      </h5>,
                    ],
                    "className": "content-section",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sabretooth",
                        "className": "ascent-name",
                      },
                      "ref": null,
                      "rendered": "Sabretooth",
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Thunder Ridge, CO",
                        "className": "ascent-location",
                      },
                      "ref": null,
                      "rendered": "Thunder Ridge, CO",
                      "type": "h5",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-times",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
